### PR TITLE
feat: Allow detection of MITM HTTPS proxies like ZScaler

### PR DESCRIPTION
**Pull request by poiru**
For security purposes, Figma heavily restrics the origins that are
allowed to load within our Electron app. Unfortunately some corporate
environments use MITM proxies like ZScaler, which intercept our
connection to `https://www.figma.com` and serve a redirect to e.g.
`https://gateway.zscloud.net` before finally redirecting back to
`https://www.figma.com`.

In order to detect this situation and allow the navigation to proceed,
we need to be able to know whether or not the certificate for our own
origin is chained to a known root. We do this by exposing
`CertVerifyResult::is_issued_by_known_root`.

If the certification verification passed without the certificate being
tied to a known root, we can safely assume that we are dealing with a
MITM proxy that has its own root certificate installed locally on the
machine and therefore loosen our origin checks.

Notes: Add `isIssuedByKnownRoot` to `ses.setCertificateVerifyProc` callback

### CODE REVIEW

**Review by MarshallOfSound — APPROVED** [review]

File: docs/api/session.md [hunk 1]
```diff
@@ -447,7 +447,8 @@ the original network configuration.
     * `hostname` String
     * `certificate` [Certificate](structures/certificate.md)
     * `validatedCertificate` [Certificate](structures/certificate.md)
-    * `verificationResult` String - Verification result from chromium.
+    * `isIssuedByKnownRoot` Boolean - `true` if Chromium recognises the root CA as a standard root. If it isn't then it's probably the case that this certificate was generated by a MITM proxy whose root has been installed locally. This is meaningless if the certificate was not trusted.
```
Comment (nornagon) [inline docs/api/session.md:450]:
```suggestion
    * `isIssuedByKnownRoot` Boolean - `true` if Chromium recognises the root CA as a standard root. If it isn't then this certificate's root has been installed locally, for example by a corporate proxy. This is meaningless if the certificate was not trusted.
```

Also, I'm not sure what "if the certificate was not trusted" here means. Something about `verificationResult`?

File: spec-main/api-session-spec.ts [hunk 2]
```diff
@@ -590,6 +590,7 @@ describe('session module', () => {
           expect(certificate.issuerCert.issuerCert.subject.commonName).to.equal('Root CA');
           expect(certificate.issuerCert.issuerCert.issuerCert).to.equal(undefined);
           expect(verificationResult).to.be.oneOf(['net::ERR_CERT_AUTHORITY_INVALID', 'net::ERR_CERT_COMMON_NAME_INVALID']);
+          expect(isIssuedByKnownRoot).to.be.false();
         };
         callback(-2);
```
Comment (Adriamo132) [inline spec-main/api-session-spec.ts:595]: Off

**Review by Adriamo132 — APPROVED** [review]

File: docs/api/session.md (hunk 1 above)
Comment (poiru) [reply docs/api/session.md:450]: @nornagon I just copied it from Chromium, but yes. How about this:

```suggestion
    * `isIssuedByKnownRoot` Boolean - `true` if Chromium recognises the root CA as a standard root. If it isn't then it's probably the case that this certificate was generated by a MITM proxy whose root has been installed locally (for example, by a corporate proxy). This should not be trusted if the `verificationResult` is not `'OK'`.
```

File: docs/api/session.md [hunk 3]
```diff
@@ -447,7 +447,8 @@ the original network configuration.
...
+    * `verificationResult` String - Verification result from Chromium.
```
Comment (poiru) [inline docs/api/session.md:451]:
```suggestion
    * `verificationResult` String - Verification result from Chromium. `OK` if the certificate is trusted.
```

**Review by nornagon — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**